src/ert/run_models/everest_run_model.py

```diff
@@ -420,44 +420,89 @@ def _on_before_forward_model_evaluation(
             logging.getLogger(EVEREST).info("User abort requested.")
             optimizer.abort_optimization()
 
+    @staticmethod
+    def _consolidate_cache_and_sim_results(
+        num_simulations: int,
+        cached_results: dict[int, Any],
+        simulated_results: NDArray[np.float64] | None = None,
+    ) -> NDArray[np.float64] | None:
+        if simulated_results is None:
+            return np.array(
+                [cached_results.get(sim) for sim in range(num_simulations)],
+                dtype=np.float64,
+            )
+        else:
+            return np.array(
+                [
+                    cached_results.get(sim, simulated_results[sim])
+                    for sim in range(num_simulations)
+                ],
+                dtype=np.float64,
+            )
+
     def _run_forward_model(
         self,
         control_values: NDArray[np.float64],
         model_realizations: list[int],
-        active_control_vectors: list[bool] | None = None,
+        active_sim_ids: list[bool] | None = None,
     ) -> tuple[NDArray[np.float64], NDArray[np.float64] | None, list[int]]:
         # Reset the current run status:
         self._status = None
 
         # Get cached_results:
-        cached_results = self._get_cached_results(control_values, model_realizations)
+        cached_objectives, cached_constraints = self._get_cached_results_by_sim_id(
+            control_values, model_realizations
+        )
```

Comment:
This does not take `active_sim_ids` into account. This means that the cache may contain values that should not be used. Or maybe they do not contain those values, but since these are also not evaluated, `_consolidate_cache_and_sim_results` might fail if it tries to construct a result from cached results only (there will be too few results). Solution would be not to cache anything that has a `False` entry in `active_sim_ids`. `_consolidate_cache_and_sim_results` would need to be adapted to make sure that the results have the right size and have zero entries in the correspoding places.

Reply:
I think you can do `control_values[active_sim_ids, :]` to filter out the unwanted values.

Reply:
need to filter `model_realizations` too, I guess, so maybe not convient.